test/e2e/nftables_test.go

```diff
@@ -23,61 +24,79 @@ var (
 	chainName      = "chain OPENSHIFT"
 )
 
+const (
+	waitDuration = 2 * time.Minute
+	mcTimeout    = 20 * time.Minute
+)
+
 var _ = Describe("Nftables", func() {
 	It("should apply firewall by blocking all ports except the ones OCP is listening on", func() {
 		masterMat, workerMat := commatrix.SeparateMatrixByRole()
-		var workerNFT []byte
+		nodeRoleToNFTables := make(map[string][]byte)
 
 		By("Creating NFT output for each role")
-		masterNFT, err := masterMat.ToNFTables()
-		Expect(err).NotTo(HaveOccurred())
-		if !isSNO {
-			workerNFT, err = workerMat.ToNFTables()
+		for _, node := range nodeList.Items {
+			role, err := types.GetNodeRole(&node)
 			Expect(err).NotTo(HaveOccurred())
 
-			if extraNFTablesFile != "" {
-				workerNFT, err = AddPortsToNFTables(workerNFT, extraNFTablesFile)
-				Expect(err).NotTo(HaveOccurred())
+			if _, exists := nodeRoleToNFTables[role]; !exists {
+				var nftableConfig []byte
+
+				if role == workerNodeRole {
+					nftableConfig, err = workerMat.ToNFTables()
+					Expect(err).NotTo(HaveOccurred())
+
+					if extraNFTablesFile != "" {
+						nftableConfig, err = AddPortsToNFTables(nftableConfig, extraNFTablesFile)
+						Expect(err).NotTo(HaveOccurred())
+					}
+				} else {
+					nftableConfig, err = masterMat.ToNFTables()
+					Expect(err).NotTo(HaveOccurred())
+
+					if extraNFTablesFile != "" {
```

Comment:
Can be done onde

Reply:
we will have a fix for master and worker this is why i kept it

Reply:
that will nit be in the loop and doing that all the time becouse of that
exists := nodeRoleToNFTables[role]; !exists
if not exist just we will do it